### ISSUE TITLE
MD025 (multiple top level headings) triggered by Front Matter config

### ISSUE THREAD

**Issue by mundodisco8** [markdownlint/markdownlint#382]
## Description
When the `Title` element is used in the Front Matter section of my markdown file, rule MD025 triggers in the first level one header of my doc 

### Environment

Markdown lint 0.42.1 in VS Code 1.57.1

### Expected Behavior

The Front Matter section of the document should not affect the processing of the linting rules of my markdown file (unless there are rules specific about Front Matter).

### Actual Behavior

When I add a "Title" Front Matter element (That I later use to get the document title into each printed page's header), my level 1 header (h1, `#`) triggers rule MD025, but there are no other h1 levels in my doc.

## Replication Case

This markdown document triggers it for me
```md
---
title: "hey"
---

# Doc Title

some test

```

Here's a screenshot

![image](https://user-images.githubusercontent.com/8402707/124390772-f110db00-dce4-11eb-878d-ce45c3477474.png)

**Comment by DavidAnson** [comment]
This would be an issue for the Node implementation of markdownlint, not the Ruby one. Specifically, the VS Code extension: https://github.com/DavidAnson/vscode-markdownlint

The behavior you describe is by design and explained by the documentation for MD025: https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md025

**Comment by mundodisco8** [comment]
Oh! I see, I didn't read the rule in this particular instance because I was quite sure that it was a bug, but after reading it it makes perfect sense! I removed the check for the front matter YAML. Thanks!
